### PR TITLE
Application Private Events

### PR DESCRIPTION
**Pull request by Mhdez221993**
In this  project we created:

1.  a new private-events Rails app and GitHub repo using Rails command line
2. A descriptive README and link to this project
3. Event and User model
4. association between the event creator (a User) and the event
5.  Controllers and corresponding routes

Set up devise to handle authentication

### CODE REVIEW

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      <% else %>
+    
+      <li class="nav-item active">
+        <%= link_to 'sing_in', new_user_session_path, class:'nav-link' %>
+      </li>       
+
+       <li class="nav-item active">
+        <%= link_to 'sing_up', new_user_registration_path, class:'nav-link' %>
+      </li>
+      <% end %>
```
Comment (SajjadAhmad14) [inline app/views/layouts/_header.html.erb:26]: Views should be separate from logic. [OPTIONAL]

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+<h3>Event User will Attent</h3>
+<% @user.attendeds.each do |event| %>
+    <% if event.date >  Time.now %>
+        <ul>
+            <li><%= event.title %></li>
+            <li><%= event.description %></li>
+            <li><%= event.date %></li>
+        </ul>
+    <% end %>
```
Comment (SajjadAhmad14) [inline app/views/users/show.html.erb]: Views should be separate from logic.  [OPTIONAL]

File: app/views/devise/mailer/email_changed.html.erb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @email %>!</p>
+
+<% if @resource.try(:unconfirmed_email?) %>
+  <p>We're contacting you to notify you that your email is being changed to <%= @resource.unconfirmed_email %>.</p>
+<% else %>
+  <p>We're contacting you to notify you that your email has been changed to <%= @resource.email %>.</p>
+<% end %>
```
Comment (SajjadAhmad14) [inline app/views/devise/mailer/email_changed.html.erb:7]: Views should be separate from logic.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [ ] You have mastered Git Flow.
- [ ] README is professional and includes the necessary info to run the project locally.
## To Improve 🙏
- [x] Please do not repeat one commit message for multiple commits. It does not look descriptive enough.
- [x] Events controller should have past events, i.e. @past_events = Event.past_events
- [x] Events controller should have future_events, i.e. @future_events = Event.future_events
- [x] You should use model scopes to create @past_events, and @future_events.
- [x] User show action should list events created by user, i.e. @created_events = current_user.events.
- [x] User show action should use event model scopes to separate past attended events and events that are upcoming.
- [x] Please add unit tests for model validations and associations.
- [x] Please add authentication tests (Rspec + Capybara).
- [x] Please add integration tests for event management (Rspec + Capybara).
- [x] Some specs are failing. KIndly fix them.
- [x] Please see inline comments for reference.
# Optional
- [ ] There should be no logic in views. Please use the helpers directory to define views helpers that use logic and render them in views.
### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Review by carlos-ssh — CHANGES_REQUESTED** [review]
**Hello team! 
your project is really nice!.**

📝**A few changes requested** ‼️:
- Would be great to have instructions for additional installations like: ```rails webpacker:install``` for non-technical people.
Screen Shot 2021-06-15 at 08 36 13

- Kindly remove unused files like json.jbuilder.
- lists events previoulsy attendented by user using event model scope, i.e.``` @past_events = current_user.attended_events.previous ```.
- lists upcoming events previously attended by user using event model scope, i.e. ```@upcoming_events = current_user.attended_events.upcoming ```
- When i try to run tests i have failed examples: ```rspec ./spec/authentication/login_spec.rb:8 # User signs in signs in @user with correct credentials
rspec ./spec/authentication/login_spec.rb:17 # User signs in doesn's sign in user with unregistered account
rspec ./spec/authentication/login_spec.rb:27 # User signs in doesn's sign in user with invalid password
rspec ./spec/authentication/login_spec.rb:37 # User signs in doesn's sign in user with invalid email```

**_Happy Coding!_** 👾👾
@carlos-ssh

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED :green_circle: :100: 
---------------
Hello @Mhdez221993 @1v4n4 Great job you did in this project :clap: :+1: 
good job working on the required changes :100: :smile: 
-------------
Now you can merge this branch :1st_place_medal: 
HAPPY CODING !!! :100:

**Pull request merged**